### PR TITLE
Move test helpers to test/helpers

### PR DESCRIPTION
**Pull request by blowery**
Fix up tests that used them
Fix config test to use useMockery, which is was not.

### CODE REVIEW

**Comment by blowery** [comment]
Adding readme's, then this should be good to go

File: test/test/helpers/nock-control/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+We wrap all tests in the single test runner with nock to disable network access.
+On occasion, we require network access as part of a test. This package includes one
+function, `allowNetworkAccess`, which will enable network access for the duration of the describe block.
+
+## Usage
+
+```js
+import { allowNetworkAccess } from 'test/helpers/nock-control';
+
+describe('my test suite', () = {
```
Comment (aduth) [inline test/test/helpers/nock-control/README.md]: Some invalid syntax and a few style issues here:

- `() = {` should be `() => {`
- `describe('` should be `describe( '`
- `it ( 'should` should be `it( 'should`

File: test/test/helpers/use-mockery/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+describe('my test suite', () = {
+
+	before( () => {
+		// THINGS REQUIRED HERE WILL NOT HAVE MOCKED DEPENDENCIES
+	} );
+
+	useMockery();
+
+	before( () => {
+		// you can require things that have mocked dependencies now!
```
Comment (aduth) [inline test/test/helpers/use-mockery/README.md]: When should I be doing the mocking? Should we include example `mockery.registerMock` or `mockery.registerSubstitute` ?

**Comment by aduth** [comment]
Aside from few documentation nitpicks, this looks good to me.

Aside, I kinda wish we didn't need nested `test/test` directory, while still being able to include as e.g. `test/helpers/nock-control` in usage.

**Comment by blowery** [comment]
> Aside, I kinda wish we didn't need nested test/test directory, while still being able to include as e.g. test/helpers/nock-control in usage.

I know. The joys using overriding NODE_PATH right there.

**Pull request merged**